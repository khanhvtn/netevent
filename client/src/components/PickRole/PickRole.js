import React from 'react';
import { Container, Grid, Typography } from '@material-ui/core';
import adminIcon from '../../images/admin-color-icon.png';
import reviewerIcon from '../../images/reviewer-color-icon.png';
import creatorIcon from '../../images/creator-color-icon.png';
import memberIcon from '../../images/member-color-icon.png';
import { useSelector } from 'react-redux';

//import makeStyles in the last
import makeStyles from './styles';
import RoleCard from './RoleCard/RoleCard';
import DashboardNavbar from '../../Layouts/Dashboard/DashboardNavbar/DashboardNavbar';

/* Default path is used to enable the first actions in user dashboard based on their role */
const roleInfo = {
    1: {
        roleName: 'Admin',
        icon: adminIcon,
        defaultPath: 'dashboard/user',
    },
    2: {
        roleName: 'Reviewer',
        icon: reviewerIcon,
        defaultPath: 'dashboard/event-reviewer',
    },
    3: {
        roleName: 'Creator',
        icon: creatorIcon,
        defaultPath: 'dashboard/event-management',
    },
    4: {
        roleName: 'Team Member',
        icon: memberIcon,
        defaultPath: 'dashboard/task',
    },
};

const PickRole = () => {
    const css = makeStyles();
    const { userInfo } = useSelector((state) => ({
        userInfo: state.user.user,
    }));
    const renderRoleCard = userInfo.role.map((roleNum, index) => {
        const { roleName, icon, defaultPath } = roleInfo[roleNum];
        return (
            <RoleCard
                key={index}
                roleNum={roleNum}
                roleName={roleName}
                icon={icon}
                delay={(index + 1) * 600}
                defaultPath={defaultPath}
            />
        );
    });
    return (
        <div className={css.main}>
            <Container maxWidth="lg" style={{ margin: 'auto' }}>
                <DashboardNavbar onPickRole={true} />
                <Typography
                    style={{ textAlign: 'center', marginBottom: '30px' }}
                    variant="h4"
                >
                    Choose your role to continue...
                </Typography>

                <Grid
                    item
                    container
                    justify="center"
                    alignItems="center"
                    spacing={3}
                >
                    {renderRoleCard}
                </Grid>
            </Container>
        </div>
    );
};

export default PickRole;
