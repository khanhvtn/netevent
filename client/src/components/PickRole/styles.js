import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(() => ({
    main: {
        minHeight: '100vh',
        display: 'flex',
    },
}));