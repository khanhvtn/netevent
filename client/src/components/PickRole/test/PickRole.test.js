import React from 'react'
import { Provider } from 'react-redux'
import { renderer } from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import beforeEach from 'jest-wrap'
import expect from 'expect'
import toJson from 'enzyme-to-json'

import PickRole from '../PickRole'

const mockStore = configureStore([])
describe('PickRole', () => {
    let store;
    let component;

    beforeEach(() => {
        store = mockStore({
            myState: 'PickRole'
        })

        store.dispatch = jest.fn()

        component = renderer.create(
            <Provider store={store}>
                <PickRole />
            </Provider>
        )
    })
    
    it('should render with given state from Redux store', () => {
        expect(toJson(component)).toMatchSnapshot()
    })
})