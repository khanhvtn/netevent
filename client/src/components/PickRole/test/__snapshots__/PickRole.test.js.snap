// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PickRole should render with given state from Redux store 1`] = `undefined`;
